tests/benchdnn/graph/ref_partition.cpp

```diff
@@ -214,7 +218,16 @@ void ref_partition_t::exec_ops(res_t *res) {
             const auto &lt = op.in_lts_[i];
             int arg = get_prim_arg_name_from_graph_op_input_offset(
                     ref_prim->get_kind(), i, use_dst);
-            ref_prim->replace_arg(arg, lt_id_2_mems_.at(lt.id_));
+            if (cond_accept_parent_output && i == 0) {
+                // Since select primitive implementation only support s8 data
+                // type for the condition input, need to transfer the f32
+                // results from the previous op to s8.
+                dnn_mem_t &dst_i
+                        = const_cast<dnn_mem_t &>(ref_prim->get_arg(arg));
+                const auto &src_i = lt_id_2_mems_.at(lt.id_);
+                SAFE_V(dst_i.reorder(src_i));
+            } else
+                ref_prim->replace_arg(arg, lt_id_2_mems_.at(lt.id_));
```

Comment:
```suggestion
            if (select_op_cond_has_parent && i == 0) {
                // Since select primitive implementation only supports s8 data
                // type for the condition input, need to transfer the f32
                // results from the previous op to s8.
                dnn_mem_t &dst_i
                        = const_cast<dnn_mem_t &>(ref_prim->get_arg(arg));
                const auto &src_i = lt_id_2_mems_.at(lt.id_);
                SAFE_V(dst_i.reorder(src_i));
                continue;
            }

            ref_prim->replace_arg(arg, lt_id_2_mems_.at(lt.id_));
```